src/js/views/Projects/Projects.jsx

```diff
@@ -76,6 +77,21 @@ function Projects() {
     )
   }
 
+  // Users can use the <ProjectSearch> component to change the filter
+  // The component changes the `f` parameter via `navigate(...)`. However,
+  // if we are already on the projects page, then the `navigate()` function
+  // does not rerender the page. We can catch this with the `useLocation()`
+  // hook and update the filter appropriately.
+  useEffect(() => {
+    if (searchParams.get('f') && state.filter !== searchParams.get('f')) {
+      setState((prevState) => ({
+        ...prevState,
+        filter: searchParams.get('f') || '',
+        refresh: true
+      }))
```

Comment:
I think this can be the body of the `onRefresh` function just below it, and then you can call `onRefresh` here. Setting data to `[]` is kind of optional, since that just means it is empty for a brief period till it gets the new data, could keep it or drop it.

Reply:
Indeed it can be combined -- 45333ed